Java/src/main/java/models/ExperimentalTypes.java

```diff
@@ -1,5 +1,5 @@
 package models;
 
 public enum ExperimentalTypes {
```

Comment:
1. You need some consistency in terms of whether it is a `Type` or `Types` as you have `ExperimentalPlaneTypes` and `MilitaryPlaneType` 
2. To simplify the whole thing, you can introduce the interface, e.g. `PlaneType` and implement it here and in `MilitaryPlaneType` and then have a common field in `Plane` class as well as common constructor there.